smack-websocket/src/main/java/org/jivesoftware/smack/websocket/implementations/AbstractWebsocketImpl.java

```diff
@@ -0,0 +1,58 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket.implementations;
+
+public abstract class AbstractWebsocketImpl implements WebsocketInterface {
+
+    protected enum WebsocketConnectionPhase {
+        readyToSendOpen,
+        openFrameSent,
+        readyToReceiveFeatures,
+        exchangingTopLevelStreamElements,
+        closingHandshake
+    }
+
+    protected String getStreamFromOpenElement(String openElement) {
+        String streamElement = openElement.replace("open", "stream")
```

Comment:
Should be
`replace("<open ", "<stream ")`
or maybe even better
`replaceFirst("\A<open ", "<stream ")`